agents-api/agents_api/workflows/task_execution/__init__.py

```diff
@@ -168,7 +168,7 @@ async def eval_step_exprs(self, step_type: WorkflowStep):
                         output = i
                         break
             case ToolCallStep(arguments=arguments):
-                tools: list[Tool] = self.context.tools
+                tools: list[Tool] = await self.context.tools
```

Comment:
we should prolly change this getter to an async method, this naming is v confusing... (but in a future pr; so let's create an issue for it)